test/mitmproxy/addons/test_maplocal.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+
+from mitmproxy.addons.maplocal import MapLocal, file_candidates
+from mitmproxy.test import taddons
+from mitmproxy.test import tflow
+
+from mitmproxy.addons.modifyheaders import parse_modify_spec
+
+
+@pytest.mark.parametrize(
+    "url,spec,expected_candidates", [
+        (
+            "https://example.org/img/topic/subtopic/test.jpg",
+            ":example.com/foo:/tmp",
+            ["/tmp/img/topic/subtopic/test.jpg", "/tmp/img/topic/test.jpg", "/tmp/img/test.jpg", "/tmp/test.jpg"]
+        ),
+        (
+            "https://example.org/img/topic/subtopic/",
+            ":/img:/tmp",
```

Comment:
I'm really not convinced that this form of patterns make much sense, same goes for the one below. What's the use case here? IMO the main point here is to specify URL prefixes, mapping all URLs that match "org" onto a folder seems really far fetched. Am I missing something?

Reply:
I agree, I didn't use particular useful or common use-case patterns, but just some that test the functionality.
Will replace them with more useful patterns, e.g., the ones I used in the docs.